v4/newrelic/sql_driver_test.go

```diff
@@ -0,0 +1,295 @@
+	txn := app.StartTransaction("hello")
+	conn, _ := connector.Connect(nil)
+	ctx := NewContext(context.Background(), txn)
+	conn.(driver.ExecerContext).ExecContext(ctx, "myoperation,mycollection", nil)
+	txn.End()
+	app.ExpectMetrics(t, driverTestMetrics)
+	app.ExpectSpanEvents(t, driverTestSpans)
+}
+
+func TestConnectorToDriver(t *testing.T) {
+	// Test that driver.Connector.Driver returns an instrumented Driver.
+	app := newTestApp(t)
+	connector := InstrumentSQLConnector(testConnector{}, testBuilder)
+	txn := app.StartTransaction("hello")
+	dr := connector.Driver()
+	conn, _ := dr.Open("myhost,myport,mydatabase")
+	ctx := NewContext(context.Background(), txn)
+	conn.(driver.ExecerContext).ExecContext(ctx, "myoperation,mycollection", nil)
+	txn.End()
+	app.ExpectMetrics(t, driverTestMetrics)
+	app.ExpectSpanEvents(t, driverTestSpans)
+}
+
+type testConnectorErr struct {
+	testConnector
+}
+
+func (c testConnectorErr) Connect(context.Context) (driver.Conn, error) { return testConnErr{}, nil }
+
+type testConnErr struct {
+	testConn
+}
+
+func (c testConnErr) QueryContext(context.Context, string, []driver.NamedValue) (driver.Rows, error) {
+	return nil, driver.ErrSkip
+}
+
+func (c testConnErr) ExecContext(context.Context, string, []driver.NamedValue) (driver.Result, error) {
+	return nil, driver.ErrSkip
+}
+
+// Ensure that if the driver used returns driver.ErrSkip that spans still have correct parentage
+func TestExecContextErrSkipReturned(t *testing.T) {
+	app := newTestApp(t)
+	connector := InstrumentSQLConnector(testConnectorErr{}, testBuilder)
+	txn := app.StartTransaction("hello")
+	conn, _ := connector.Connect(nil)
+	ctx := NewContext(context.Background(), txn)
+	conn.(driver.ExecerContext).ExecContext(ctx, "myoperation,mycollection", nil)
+	txn.StartSegment("second").End()
+	txn.End()
+	app.ExpectSpanEvents(t, []internal.WantSpan{
+		{
+			Name:       "second",
+			SpanID:     "0000000000000003",
+			TraceID:    "00000000000000020000000000000000",
+			ParentID:   "0000000000000002",
+			Attributes: map[string]interface{}{},
+		},
+		{
+			Name:       "hello",
+			SpanID:     "0000000000000002",
+			TraceID:    "00000000000000020000000000000000",
+			ParentID:   "0000000000000000",
+			Attributes: map[string]interface{}{},
+		},
+	})
+}
+
+// Ensure that if the driver used returns driver.ErrSkip that spans still have correct parentage
+func TestQueryContextErrSkipReturned(t *testing.T) {
+	app := newTestApp(t)
+	connector := InstrumentSQLConnector(testConnectorErr{}, testBuilder)
+	txn := app.StartTransaction("hello")
+	conn, _ := connector.Connect(nil)
+	ctx := NewContext(context.Background(), txn)
+	conn.(driver.QueryerContext).QueryContext(ctx, "myoperation,mycollection", nil)
+	txn.StartSegment("second").End()
+	txn.End()
+	app.ExpectSpanEvents(t, []internal.WantSpan{
+		{
+			Name:       "second",
+			SpanID:     "0000000000000003",
+			TraceID:    "00000000000000020000000000000000",
+			ParentID:   "0000000000000002",
```

Comment:
It was through updating this test that I discovered we'd forgotten the update to the `sql_driver.go` file! Yay for tests!